### PR TITLE
feat: Adding pivot point standard impls

### PR DESCRIPTION
**Pull request by dandxy89**
To implement this I needed to increase `IndicatorResult` result size to `12`. 

Resolves: https://github.com/amv-dev/yata/issues/36

### CODE REVIEW

**Comment by dandxy89** [comment]
@amv-dev - can you take a look at this and advise on changes?

**Comment by amv-dev** [comment]
@dandxy89 Hello! Sorry for this looooooong anwer. I was having a lot of busy days...
I don't think it is good idea to increase size for `IndicatorResult`. The whole idea of fixed size for `IndicatorResult` was totally wrong and I already working on next version of the library, where this issue is handled correctly. 

But for now I'd recommend you to make you indicator as `Method`, not an `Indicator`. `Method` allows to define it's own `Output` type, where all the signals may be described.

**Comment by dandxy89** [comment]
No problem @amv-dev. 

Thats for the feedback - much appreciated.

Let me re-familiarise myself with my changes and I'll update the PR over the next coming days.

**Comment by dandxy89** [comment]
Just as an FYI this PR only contains a impl for Traditional. 

```
Traditional <- Adding in this PR
Fibonacci
Woodie
Classic
DM
Camarilla
```

I can follow up with another PR to implement the [remainder](https://www.tradingview.com/support/solutions/43000521824-pivot-points-standard/)

File: src/methods/pivot_point_standard.rs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+/// The Pivot Point indicator is calculated based on the high, low, and close prices of the previous trading
+/// period (such as a day, week, or month) and provides several support and resistance levels.
+///
+/// ## Links
+///
+/// * <https://en.wikipedia.org/wiki/Pivot_point_(technical_analysis)>
+///
+pub struct PivotPointStandard {
+	close: ValueType,
```
Comment (amv-dev) [inline src/methods/pivot_point_standard.rs:20]: Do we even need internal state? It seems like `PivotPointStandard` may be simple unit struct without internal state.

File: src/methods/pivot_point_standard.rs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+///
+/// # Output type
+///
+/// Output type is [`PivotPointTraditionalOutput`]
+///
+/// # Links
+///
+/// * <https://en.wikipedia.org/wiki/Pivot_point_(technical_analysis)>
+///
+pub struct PivotPointStandard {}
```
Comment (amv-dev) [inline src/methods/pivot_point_standard.rs:31]: Please remove those curly brackets. It's useless.

**Review by amv-dev — CHANGES_REQUESTED** [review]

**Comment by amv-dev** [comment]
Please make this change and I will merge your PR.

But I think this feature must be refactored in the future. As I see, there is no internal state. On the other hand there are too many values in the output. So I think the whole feature might be created by traits. Something like
```rust
pub trait PivotPointStandard: OHLCV {
  fn pp(&self) -> ValueType {
    (self.high() + self.low() + self.close())/3.0
  }
  ...
}

impl<T: OHLCV> PivotPointStandard for T {}
```

**Comment by dandxy89** [comment]
Thanks for the feedback. 

I'll look at the suggestions later this year.

**Pull request closed without merging**